src/reducers/flexColumnWidths.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright Schrodinger, LLC
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule flexColumnWidths
+ */
+
+'use strict';
+
+import scrollbarsVisibleSelector from '../selectors/scrollbarsVisible';
+import { getColumn, updateColWidth } from './updateColWidth';
+
+/**
+ * @param state
+ * @param props
+ */
+export function initializeFlexColumnWidths(state) {
+  const { scrollEnabledY } = scrollbarsVisibleSelector(state);
+  const tableWidth = state.tableSize.width;
+  const scrollbarSpace = scrollEnabledY ? state.scrollbarYWidth : 0;
+  let availableWidth = tableWidth - scrollbarSpace - state.fixedContentWidth;
+
+  let flexTotal = 0;
+
+  for (let column of [...state.fixedColumns, ...state.fixedRightColumns]) {
+    flexTotal += column.props.flexGrow || 0;
+  }
+
+  // iterate scrollable columns until they fill up the available space
+  let idx = 0;
+  while (
+    availableWidth > 0 &&
+    idx < state.columnSettings.scrollableColumnsCount
+  ) {
+    const column = getColumn(state, idx);
+    availableWidth -= column.props.width;
+    flexTotal += column.props.flexGrow;
+    idx++;
+  }
+
+  // check if there's free space left to fill up flex widths
+  if (
+    availableWidth <= 0 &&
+    idx < state.columnSettings.scrollableColumnsCount
+  ) {
+    return;
+  }
+
+  // generic function to update given set of columns with new width based on their flex grow.
+  const updateFlexWidths = (columns, columnUpdater) => {
```

Comment:
`updateFlexWidths` walks over the given list of columns, calculates the new width, and calls columnUpdate with the new width and column index.

This is called thrice just below, once for scrollable columns, once for fixed columns, and once for fixed right.